tests/test_graph.py

```diff
@@ -512,30 +514,100 @@ def test_function_graph_display():
         all_nodes.add(n)
     # hack of a test -- but it works... sort the lines and match them up.
     # why? because for some reason given the same graph, the output file isn't deterministic.
-    expected = sorted(
+    # for the same reason, order of input nodes are non-deterministic
+    expected_set = set(
         [
+            '\t\tfunction [fillcolor="#b4d8e4" fontname=Helvetica margin=0.15 shape=rectangle style="rounded,filled"]\n',
+            "\t\tgraph [fontname=helvetica label=Legend rank=same]\n",
+            "\t\tinput [fontname=Helvetica margin=0.15 shape=rectangle style=dashed]\n",
+            '\t\toutput [fillcolor="#FFC857" fontname=Helvetica margin=0.15 shape=rectangle style="rounded,filled"]\n',
+            "\tA -> B\n",
+            "\tA -> C\n",
+            '\tA [label=<<b>A</b><br /><br /><i>int</i>> fillcolor="#b4d8e4" fontname=Helvetica margin=0.15 shape=rectangle style="rounded,filled"]\n',
+            '\tB [label=<<b>B</b><br /><br /><i>int</i>> fillcolor="#FFC857" fontname=Helvetica margin=0.15 shape=rectangle style="rounded,filled"]\n',
+            '\tC [label=<<b>C</b><br /><br /><i>int</i>> fillcolor="#b4d8e4" fontname=Helvetica margin=0.15 shape=rectangle style="rounded,filled"]\n',
+            "\t_A_inputs -> A\n",
+            # commenting out input node: '\t_A_inputs [label=<<table border="0"><tr><td>c</td><td>int</td></tr><tr><td>b</td><td>int</td></tr></table>> fontname=Helvetica margin=0.15 shape=rectangle style=dashed]\n',
+            "\tgraph [compound=true concentrate=true rankdir=LR ranksep=0.4]\n",
+            "\tsubgraph cluster__legend {\n",
+            "\t}\n",
             "// Dependency Graph\n",
             "digraph {\n",
-            "\tA [label=A]\n",
-            "\tC [label=C]\n",
-            "\tB [label=B shape=rectangle]\n",
-            '\tc [label="Input: c" style=dashed]\n',
-            '\tb [label="Input: b" style=dashed]\n',
-            "\tb -> A\n",
-            "\tc -> A\n",
-            "\tA -> C\n",
-            "\tA -> B\n",
             "}\n",
         ]
     )
-    with tempfile.TemporaryDirectory() as tmp_dir:
-        path = tmp_dir.join("test.dot")
-        fg.display(all_nodes, str(path), {"view": False}, None, node_modifiers)
-        with open(str(path), "r") as dot_file:
-            actual = sorted(dot_file.readlines())
-            assert actual == expected
-        dot_file = fg.display(all_nodes, output_file_path=None, node_modifiers=node_modifiers)
-        assert dot_file is not None
+
+    fg.display(
+        all_nodes,
+        output_file_path=str(dot_file_path),
+        render_kwargs={"view": False},
+        node_modifiers=node_modifiers,
+    )
+
+    dot = dot_file_path.open("r").readlines()
+    dot_set = set(dot)
+
+    assert dot_set.issuperset(expected_set) and len(dot_set.difference(expected_set)) == 1
```

Comment:
Because of the sorting issues for DOT file lines and now the content of input nodes (order of rows in the table), I used sets instead. The DOT lines `expected_set` have the input node commented out. Therefore, the newly produced DOT file `dot_set` should be a superset of `expected_set` with exactly one more line for the input node.